src/python/WMCore/Storage/Backends/GFAL2Impl.py

```diff
@@ -111,37 +138,55 @@ def buildCopyCommandDict(self, sourcePFN, targetPFN, options=None, checksums=Non
         copyCommandDict['source'] = self.createFinalPFN(sourcePFN)
         copyCommandDict['destination'] = self.createFinalPFN(targetPFN)
 
+        if authMethod is None:
```

Comment:
It is not clear to me how we should deal with a third case - not explicitly asking for any authentication method. 
My understanding is that gfal will discover which credentials it can use and it the stage out will likely work, but we then lack clarity in the command and it might become harder to understand which authentication mode the job used.
Just a thought, not necessarily a request for change.

Reply:
we can make `self.setAuthX509 + " " + self.setAuthToken` for `set_auth`, to be sure the env knows eventually where the credentials are.

Reply:
done